src/server/config/env/runTime.js

```diff
@@ -52,16 +52,26 @@ const runTime = [
   {
     name: 'HTTP_PORT',
     normalize: (input) => {
-      const parsed = parseInt(input, 10);
-      // make sure the parsed value is the same value as input
-      // input may be a string or a number, we don't want === in this case, just ==
-      if (Number.isNaN(parsed) || parsed != input) { // eslint-disable-line eqeqeq
-        throw new Error(`env var HTTP_PORT needs to be a valid integer, given "${input}"`);
-      } else {
-        return parsed;
+      if (input) {
+        const parsed = parseInt(input, 10);
+        // make sure the parsed value is the same value as input
+        // input may be a string or a number, we don't want === in this case, just ==
+        if (Number.isNaN(parsed) || parsed != input) { // eslint-disable-line eqeqeq
+          throw new Error(`env var HTTP_PORT needs to be a valid integer, given "${input}"`);
+        } else {
+          return parsed;
+        }
       }
+      return undefined;
     },
-    defaultValue: () => 3000,
+    validate: (value) => {
+      if (!process.env.HTTPS_PORT && !value && !process.env.PORT) {
+        throw new Error('Either of the `HTTP_PORT` or `HTTPS_PORT` or `PORT` environment variables must be defined');
+      }
+    },
+    defaultValue: () => (process.env.NODE_ENV === 'development'
+      ? 3000
+      : undefined),
```

Comment:
`undefined` gets coerced to a string in `process.env` as you found, let's rethink this approach? what are the intentions behind this?